From 6833e76aed6ac8329562cb1d8606d8317bbf9ac2 Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Wed, 6 May 2020 21:50:31 -0700
Subject: [PATCH 1/3] RDK: Customize DTS files [1/1]

PD#SWPL-25661

Problem:
  Need to customize DTS file RDK Project

Solution:
  Customize DTS file for RDK
  AB311
  AB301
  U212

Verify:
  u212

Change-Id: I7b18402ba2a55ee80cb704fb791e5b03791e74ca
Signed-off-by: Pradeep Sriram <pradeep.sriram@amlogic.com>
---
 .../arm/boot/dts/amlogic/g12a_s905x2_u212.dts |  32 ++++-
 .../amlogic/g12a_s905x2_u212_buildroot.dts    |   2 +-
 .../g12a_s905x2_u212_drm_buildroot.dts        | 130 +++++-------------
 .../amlogic/partition_mbox_normal_P_rdk.dtsi  | 122 ++++++++++++++++
 .../boot/dts/amlogic/partition_tv_ab_rdk.dtsi | 112 +++++++++++++++
 .../tm2_revb_t962e2_ab311_drm_buildroot.dts   |  42 ++++--
 .../tm2_revb_t962x3_ab301_drm_buildroot.dts   |  40 +++++-
 .../tm2_t962e2_ab311_drm_buildroot.dts        |  40 +++++-
 .../tm2_t962x3_ab301_drm_buildroot.dts        |  40 +++++-
 9 files changed, 432 insertions(+), 128 deletions(-)
 create mode 100644 arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi
 create mode 100644 arch/arm/boot/dts/amlogic/partition_tv_ab_rdk.dtsi

diff --git a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212.dts b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212.dts
index 434e9a8813e8..7fd2584fa6fa 100644
--- a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212.dts
+++ b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212.dts
@@ -109,7 +109,8 @@
 			 * 10x4736064=45.2M(0x2e) support 12bit
 			 * 10x4074560=40M(0x28) support 10bit
 			 */
-			size = <0x02800000>;
+			//size = <0x02800000>;
+			size = <0x08800000>;//2ch
 			alignment = <0x400000>;
 		};
 		/*  POST PROCESS MANAGER */
@@ -292,14 +293,39 @@
 		dev_name = "ppmgr";
 		status = "okay";
 	};
+	di_multi_v3 {
+		compatible = "amlogic, dimv3-g12a";
+		status = "okay";
+		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
+		flag_cma = <1>;
+		//memory-region = <&di_reserved>;
+		memory-region = <&di_cma_reserved>;
+		interrupts = <0 46 1
+				0 40 1>;
+		interrupt-names = "pre_irq", "post_irq";
+		clocks = <&clkc CLKID_VPU_CLKB_TMP_COMP>,
+			<&clkc CLKID_VPU_CLKB_COMP>;
+		clock-names = "vpu_clkb_tmp_composite",
+			"vpu_clkb_composite";
+		clock-range = <334 667>;
+		/* buffer-size = <3621952>;(yuv422 8bit) */
+		//no use buffer-size = <4074560>;/*yuv422 fullpack*/
+		/* reserve-iomap = "true"; */
+		/* if enable nr10bit, set nr10bit-support to 1 */
+		post-wr-support = <1>;
+		nr10bit-support = <1>;
+		nrds-enable = <1>;
+		pps-enable = <1>;
+		ch_nub = <2>;
+	};
 
 	deinterlace {
 		compatible = "amlogic, deinterlace";
-		status = "okay";
+		status = "disable";	//"okay";
 		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
 		flag_cma = <1>;
 		//memory-region = <&di_reserved>;
-		memory-region = <&di_cma_reserved>;
+		//memory-region = <&di_cma_reserved>;
 		interrupts = <0 46 1
 				0 40 1>;
 		interrupt-names = "pre_irq", "post_irq";
diff --git a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_buildroot.dts b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_buildroot.dts
index 1fbb34d7e394..9a5fecea2b04 100644
--- a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_buildroot.dts
@@ -899,7 +899,7 @@
 		compatible = "amlogic, g12a-snd-tdmb";
 		#sound-dai-cells = <0>;
 		dai-tdm-lane-slot-mask-in = <0 1 0 0>;
-		dai-tdm-lane-slot-mask-out = <1 0 0 0>;
+		dai-tdm-lane-slot-mask-out = <1 1 1 1>;
 		dai-tdm-clk-sel = <1>;
 		clocks = <&clkaudio CLKID_AUDIO_MCLK_B
 				&clkc CLKID_MPLL1
diff --git a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts
index 50012ae3702a..c7f815f5eaf0 100644
--- a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts
@@ -19,6 +19,7 @@
 
 #include "mesong12a.dtsi"
 #include "mesong12a_drm.dtsi"
+#include "partition_mbox_normal_P_rdk.dtsi"
 
 / {
 	model = "Amlogic";
@@ -76,7 +77,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x8000000>;
+			size = <0xC800000>;
 			alignment = <0x400000>;
 			no-kernel-map;
 		};
@@ -101,7 +102,8 @@
 			 * 10x4736064=45.2M(0x2e) support 12bit
 			 * 10x4074560=40M(0x28) support 10bit
 			 */
-			size = <0x02800000>;
+			//size = <0x02800000>;
+			size = <0x08800000>;//2ch
 			alignment = <0x400000>;
 		};
 		/*  POST PROCESS MANAGER */
@@ -114,8 +116,9 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
+			alloc-ranges = <0x0 0x30000000>;
 			linux,contiguous-region;
 		};
 		/* codec shared reserved */
@@ -244,14 +247,39 @@
 		dev_name = "ppmgr";
 		status = "okay";
 	};
+	di_multi_v3 {
+		compatible = "amlogic, dimv3-g12a";
+		status = "okay";
+		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
+		flag_cma = <1>;
+		//memory-region = <&di_reserved>;
+		memory-region = <&di_cma_reserved>;
+		interrupts = <0 46 1
+				0 40 1>;
+		interrupt-names = "pre_irq", "post_irq";
+		clocks = <&clkc CLKID_VPU_CLKB_TMP_COMP>,
+			<&clkc CLKID_VPU_CLKB_COMP>;
+		clock-names = "vpu_clkb_tmp_composite",
+			"vpu_clkb_composite";
+		clock-range = <334 667>;
+		/* buffer-size = <3621952>;(yuv422 8bit) */
+		//no use buffer-size = <4074560>;/*yuv422 fullpack*/
+		/* reserve-iomap = "true"; */
+		/* if enable nr10bit, set nr10bit-support to 1 */
+		post-wr-support = <1>;
+		nr10bit-support = <1>;
+		nrds-enable = <1>;
+		pps-enable = <1>;
+		ch_nub = <2>;
+	};
 
 	deinterlace {
 		compatible = "amlogic, deinterlace";
-		status = "okay";
+		status = "disable";	//"okay";
 		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
 		flag_cma = <1>;
 		//memory-region = <&di_reserved>;
-		memory-region = <&di_cma_reserved>;
+		//memory-region = <&di_cma_reserved>;
 		interrupts = <0 46 1
 				0 40 1>;
 		interrupt-names = "pre_irq", "post_irq";
@@ -272,98 +300,10 @@
 	ionvideo {
 		compatible = "amlogic, ionvideo";
 		dev_name = "ionvideo";
-		status = "okay";
+		status = "disable";
 	};
 
 
-	partitions: partitions{
-		parts = <14>;
-		part-0 = <&logo>;
-		part-1 = <&recovery>;
-		part-2 = <&misc>;
-		part-3 = <&dto>;
-		part-4 = <&cri_data>;
-		part-5 = <&param>;
-		part-6 = <&boot>;
-		part-7 = <&rsv>;
-		part-8 = <&tee>;
-		part-9 = <&vendor>;
-		part-10 = <&odm>;
-		part-11 = <&system>;
-		part-12 = <&cache>;
-		part-13 = <&data>;
-
-		logo:logo{
-			pname = "logo";
-			size = <0x0 0x800000>;
-			mask = <1>;
-		};
-		recovery:recovery{
-			pname = "recovery";
-			size = <0x0 0x1800000>;
-			mask = <1>;
-		};
-		misc:misc{
-			pname = "misc";
-			size = <0x0 0x800000>;
-			mask = <1>;
-		};
-		dto:dto{
-			pname = "dto";
-			size = <0x0 0x800000>;
-			mask = <1>;
-		};
-		cri_data:cri_data{
-			pname = "cri_data";
-			size = <0x0 0x800000>;
-			mask = <2>;
-		};
-		rsv:rsv{
-			pname = "rsv";
-			size = <0x0 0x1000000>;
-			mask = <1>;
-		};
-		param:param{
-			pname = "param";
-			size = <0x0 0x1000000>;
-			mask = <2>;
-		};
-		boot:boot{
-			pname = "boot";
-			size = <0x0 0x1000000>;
-			mask = <1>;
-		};
-		tee:tee{
-			pname = "tee";
-			size = <0x0 0x2000000>;
-			mask = <1>;
-		};
-		vendor:vendor{
-			pname = "vendor";
-			size = <0x0 0x10000000>;
-			mask = <1>;
-		};
-		odm:odm{
-			pname = "odm";
-			size = <0x0 0x10000000>;
-			mask = <1>;
-		};
-		system:system{
-			pname = "system";
-			size = <0x0 0x80000000>;
-			mask = <1>;
-		};
-		cache:cache{
-			pname = "cache";
-			size = <0x0 0x46000000>;
-			mask = <2>;
-		};
-		data:data{
-			pname = "data";
-			size = <0xffffffff 0xffffffff>;
-			mask = <4>;
-		};
-	};
 	unifykey{
 		compatible = "amlogic, unifykey";
 		status = "ok";
@@ -886,7 +826,7 @@
 		compatible = "amlogic, g12a-snd-tdmb";
 		#sound-dai-cells = <0>;
 		dai-tdm-lane-slot-mask-in = <0 1 0 0>;
-		dai-tdm-lane-slot-mask-out = <1 0 0 0>;
+		dai-tdm-lane-slot-mask-out = <1 1 1 1 >;
 		dai-tdm-clk-sel = <1>;
 		clocks = <&clkaudio CLKID_AUDIO_MCLK_B
 				&clkc CLKID_MPLL1
diff --git a/arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi b/arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi
new file mode 100644
index 000000000000..0a2daae77346
--- /dev/null
+++ b/arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi
@@ -0,0 +1,122 @@
+﻿ /*
+  * Amlogic partition set for normal
+  *
+  * Copyright (c) 2017-2017 Amlogic Ltd
+  *
+  * This file is licensed under a dual GPLv2 or BSD license.
+  *
+  */
+
+/ {
+	partitions: partitions{
+				parts = <18>;
+				part-0 = <&dtbbak>;
+				part-1 = <&bootloaderbak>;
+				part-2 = <&recovery>;
+				part-3 = <&logo>;
+				part-4 = <&misc>;
+				part-5 = <&cri_data>;
+				part-6 = <&rsv>;
+				part-7 = <&param>;
+				part-8 = <&tee>;
+				part-9 = <&oem>;
+				part-10 = <&vbmeta_a>;
+				part-11 = <&boot_a>;
+				part-12 = <&system_a>;
+				part-13 = <&vbmeta_b>;
+				part-14 = <&boot_b>;
+				part-15 = <&system_b>;
+				part-16 = <&cache>;
+				part-17 = <&data>;
+		dtbbak:dtbbak{
+				   pname = "dtbbak";
+				   size = <0x0 0x100000>;
+				   mask = <1>;
+			   };
+		bootloaderbak:bootloaderbak{
+						  pname = "bootloaderbak";
+						  size = <0x0 0x200000>;
+						  mask = <1>;
+					  };
+		recovery:recovery{
+					 pname = "recovery";
+					 size = <0x0 0x4000000>;
+					 mask = <1>;
+				 };
+		logo:logo{
+				 pname = "logo";
+				 size = <0x0 0x800000>;
+				 mask = <1>;
+			 };
+		misc:misc{
+				 pname = "misc";
+				 size = <0x0 0x200000>;
+				 mask = <1>;
+			 };
+		cri_data:cri_data {
+					 pname = "cri_data";
+					 size = <0x0 0x400000>;
+					 mask = <2>;
+				 };
+		rsv:rsv{
+				pname = "rsv";
+				size = <0x0 0x1000000>;
+				mask = <1>;
+			};
+		param:param{
+				  pname = "param";
+				  size = <0x0 0x1000000>;
+				  mask = <2>;
+			  };
+		tee:tee{
+				pname = "tee";
+				size = <0x0 0x2000000>;
+				mask = <1>;
+			};
+		oem:oem{
+				pname = "oem";
+				size = <0x0 0x800000>;
+				mask = <1>;
+			};
+		vbmeta_a:vbmeta_a{
+					 pname = "vbmeta_a";
+					 size = <0x0 0x200000>;
+					 mask = <1>;
+				 };
+		boot_a:boot_a {
+				   pname = "boot_a";
+				   size = <0x0 0x1000000>;
+				   mask = <1>;
+			   };
+		system_a:system_a {
+					 pname = "system_a";
+					 size = <0x0 0x96000000>;
+					 mask = <1>;
+				 };
+		vbmeta_b:vbmeta_b {
+					 pname = "vbmeta_b";
+					 size = <0x0 0x200000>;
+					 mask = <1>;
+				 };
+		boot_b:boot_b {
+				   pname = "boot_b";
+				   size = <0x0 0x1000000>;
+				   mask = <1>;
+			   };
+		system_b:system_b {
+					 pname = "system_b";
+					 size = <0x0 0x96000000>;
+					 mask = <1>;
+				 };
+		cache:cache {
+				  pname = "cache";
+				  size = <0x0 0xC800000>;
+				  mask = <2>;
+			  };
+		data:data {
+		 pname = "data";
+		 size = <0xffffffff 0xffffffff>;
+		 mask = <4>;
+	 };
+			};
+};/* end of / */
diff --git a/arch/arm/boot/dts/amlogic/partition_tv_ab_rdk.dtsi b/arch/arm/boot/dts/amlogic/partition_tv_ab_rdk.dtsi
new file mode 100644
index 000000000000..6e2f8b7622f7
--- /dev/null
+++ b/arch/arm/boot/dts/amlogic/partition_tv_ab_rdk.dtsi
@@ -0,0 +1,112 @@
+﻿ /*
+  * Amlogic partition set for normal
+  *
+  * Copyright (c) 2017-2017 Amlogic Ltd
+  *
+  * This file is licensed under a dual GPLv2 or BSD license.
+  *
+  */
+
+/ {
+	partitions: partitions{
+				parts = <16>;
+				part-0 = <&recovery>;
+				part-1 = <&logo>;
+				part-2 = <&misc>;
+				part-3 = <&cri_data>;
+				part-4 = <&rsv>;
+				part-5 = <&param>;
+				part-6 = <&tee>;
+				part-7 = <&vendor>;
+				part-8 = <&vbmeta_a>;
+				part-9 = <&boot_a>;
+				part-10 = <&system_a>;
+				part-11 = <&vbmeta_b>;
+				part-12 = <&boot_b>;
+				part-13 = <&system_b>;
+				part-14 = <&user_apps>;
+				part-15 = <&data>;
+		recovery:recovery{
+					 pname = "recovery";
+					 size = <0x0 0x4000000>;
+					 mask = <1>;
+				 };
+		logo:logo{
+				 pname = "logo";
+				 size = <0x0 0x800000>;
+				 mask = <1>;
+			 };
+		misc:misc{
+				 pname = "misc";
+				 size = <0x0 0x200000>;
+				 mask = <1>;
+			 };
+		cri_data:cri_data {
+					 pname = "cri_data";
+					 size = <0x0 0x400000>;
+					 mask = <2>;
+				 };
+		rsv:rsv{
+				pname = "rsv";
+				size = <0x0 0x1000000>;
+				mask = <1>;
+			};
+		param:param{
+				  pname = "param";
+				  size = <0x0 0x1000000>;
+				  mask = <2>;
+			  };
+		tee:tee{
+				pname = "tee";
+				size = <0x0 0x2000000>;
+				mask = <1>;
+			};
+		vendor:vendor
+		{
+			pname = "vendor";
+			size = <0x0 0x14000000>;
+			mask = <1>;
+		};
+
+		vbmeta_a:vbmeta_a{
+					 pname = "vbmeta_a";
+					 size = <0x0 0x200000>;
+					 mask = <1>;
+				 };
+		boot_a:boot_a {
+				   pname = "boot_a";
+				   size = <0x0 0x1000000>;
+				   mask = <1>;
+			   };
+		system_a:system_a {
+					 pname = "system_a";
+					 size = <0x0 0x96000000>;
+					 mask = <1>;
+				 };
+		vbmeta_b:vbmeta_b {
+					 pname = "vbmeta_b";
+					 size = <0x0 0x200000>;
+					 mask = <1>;
+				 };
+		boot_b:boot_b {
+				   pname = "boot_b";
+				   size = <0x0 0x1000000>;
+				   mask = <1>;
+			   };
+		system_b:system_b {
+					 pname = "system_b";
+					 size = <0x0 0x96000000>;
+					 mask = <1>;
+				 };
+		user_apps:user_apps{
+					pname = "user_apps";
+					size = <0x0 0x80000000>;
+					mask = <1>;
+				};
+		data:data {
+		 pname = "data";
+		 size = <0xffffffff 0xffffffff>;
+		 mask = <4>;
+	 };
+			};
+};/* end of / */
diff --git a/arch/arm/boot/dts/amlogic/tm2_revb_t962e2_ab311_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/tm2_revb_t962e2_ab311_drm_buildroot.dts
index 4bcc7426c787..20770a74681a 100644
--- a/arch/arm/boot/dts/amlogic/tm2_revb_t962e2_ab311_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/tm2_revb_t962e2_ab311_drm_buildroot.dts
@@ -19,7 +19,7 @@
 
 #include "mesontm2_revb.dtsi"
 #include "mesontm2_drm.dtsi"
-#include "partition_mbox_normal_P_32.dtsi"
+#include "partition_tv_ab_rdk.dtsi"
 
 / {
 	model = "Amlogic TM2 REVB T962E2 AB311";
@@ -82,10 +82,10 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
 			linux,contiguous-region;
-			alloc-ranges = <0x31000000 0x13400000>;
+			alloc-ranges = <0x0 0x30000000>;
 		};
 
 		/* codec shared reserved */
@@ -99,7 +99,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x8000000>;
+			size = <0xC800000>;
 			alignment = <0x400000>;
 			no-kernel-map;
 		};
@@ -150,7 +150,8 @@
 			 * 10x4736064=45.2M(0x2e) support 12bit
 			 * 10x4074560=40M(0x28) support 10bit
 			 */
-			size = <0x02800000>;
+			//size = <0x02800000>;
+			size = <0x08800000>;//2ch
 			alignment = <0x400000>;
 		};
 
@@ -208,8 +209,8 @@
 		status = "okay";
 	};
 
-	deinterlace {
-		compatible = "amlogic, deinterlace_di_tm2b";
+	di_multi_v3 {
+		compatible = "amlogic, dimv3-g12a";
 		status = "okay";
 		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
 		flag_cma = <1>;
@@ -224,6 +225,31 @@
 			"vpu_clkb_composite";
 		clock-range = <334 667>;
 		/* buffer-size = <3621952>;(yuv422 8bit) */
+		//no use buffer-size = <4074560>;/*yuv422 fullpack*/
+		/* reserve-iomap = "true"; */
+		/* if enable nr10bit, set nr10bit-support to 1 */
+		post-wr-support = <1>;
+		nr10bit-support = <1>;
+		nrds-enable = <1>;
+		pps-enable = <1>;
+		ch_nub = <2>;
+	};
+	deinterlace {
+		compatible = "amlogic, deinterlace";
+		status = "disable";
+		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
+		flag_cma = <1>;
+		//memory-region = <&di_reserved>;
+		memory-region = <&di_cma_reserved>;
+		interrupts = <0 46 1
+				0 40 1>;
+		interrupt-names = "pre_irq", "post_irq";
+		clocks = <&clkc CLKID_VPU_CLKB_TMP_COMP>,
+			<&clkc CLKID_VPU_CLKB_COMP>;
+		clock-names = "vpu_clkb_tmp_composite",
+			"vpu_clkb_composite";
+		clock-range = <334 667>;
+		/* buffer-size = <3621952>;(yuv422 8bit) */
 		buffer-size = <4074560>;/*yuv422 fullpack*/
 		/* reserve-iomap = "true"; */
 		/* if enable nr10bit, set nr10bit-support to 1 */
@@ -1823,7 +1849,7 @@
 	//pinctrl-names = "internal_eth_pins", "internal_gpio_pins";
 	//pinctrl-0 = <&internal_eth_pins>;
 	//pinctrl-1 = <&internal_gpio_pins>;
-	mc_val = <0x4be04>;
+	mc_val = <0x49e04>;
 
 	internal_phy=<1>;
 };
diff --git a/arch/arm/boot/dts/amlogic/tm2_revb_t962x3_ab301_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/tm2_revb_t962x3_ab301_drm_buildroot.dts
index 3a2a51cfb0f5..76e258ca9530 100644
--- a/arch/arm/boot/dts/amlogic/tm2_revb_t962x3_ab301_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/tm2_revb_t962x3_ab301_drm_buildroot.dts
@@ -19,7 +19,7 @@
 
 #include "mesontm2_revb.dtsi"
 #include "mesontm2_drm.dtsi"
-#include "partition_mbox_normal_P_32.dtsi"
+#include "partition_tv_ab_rdk.dtsi"
 #include "mesontm2_revb_t962x3_ab301-panel.dtsi"
 
 / {
@@ -91,10 +91,10 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
 			linux,contiguous-region;
-			alloc-ranges = <0x30000000 0x50000000>;
+			alloc-ranges = <0x0 0x30000000>;
 		};
 
 		/* codec shared reserved */
@@ -108,7 +108,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x2000000>;
+			size = <0xC800000>;
 			alignment = <0x400000>;
 			no-kernel-map;
 		};
@@ -159,7 +159,8 @@
 			 * 10x4736064=45.2M(0x2e) support 12bit
 			 * 10x4074560=40M(0x28) support 10bit
 			 */
-			size = <0x02800000>;
+			//size = <0x02800000>;
+			size = <0x08800000>;//2ch
 			alignment = <0x400000>;
 		};
 
@@ -216,9 +217,34 @@
 		status = "okay";
 	};
 
+	di_multi_v3 {
+		compatible = "amlogic, dimv3-g12a";
+		status = "okay";
+		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
+		flag_cma = <1>;
+		//memory-region = <&di_reserved>;
+		memory-region = <&di_cma_reserved>;
+		interrupts = <0 46 1
+				0 40 1>;
+		interrupt-names = "pre_irq", "post_irq";
+		clocks = <&clkc CLKID_VPU_CLKB_TMP_COMP>,
+			<&clkc CLKID_VPU_CLKB_COMP>;
+		clock-names = "vpu_clkb_tmp_composite",
+			"vpu_clkb_composite";
+		clock-range = <334 667>;
+		/* buffer-size = <3621952>;(yuv422 8bit) */
+		//no use buffer-size = <4074560>;/*yuv422 fullpack*/
+		/* reserve-iomap = "true"; */
+		/* if enable nr10bit, set nr10bit-support to 1 */
+		post-wr-support = <1>;
+		nr10bit-support = <1>;
+		nrds-enable = <1>;
+		pps-enable = <1>;
+		ch_nub = <2>;
+	};
 	deinterlace {
 		compatible = "amlogic, deinterlace_di_tm2b";
-		status = "okay";
+		status = "disable";
 		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
 		flag_cma = <1>;
 		//memory-region = <&di_reserved>;
@@ -1864,7 +1890,7 @@
 	pinctrl-names = "internal_eth_pins", "internal_gpio_pins";
 	pinctrl-0 = <&internal_eth_pins>;
 	pinctrl-1 = <&internal_gpio_pins>;
-	mc_val = <0x4be04>;
+	mc_val = <0x49e04>;
 
 	internal_phy=<1>;
 };
diff --git a/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts
index bb6b0476f880..3f89b5b9e692 100644
--- a/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts
@@ -19,7 +19,7 @@
 
 #include "mesontm2.dtsi"
 #include "mesontm2_drm.dtsi"
-#include "partition_mbox_normal_P_32.dtsi"
+#include "partition_tv_ab_rdk.dtsi"
 
 / {
 	model = "Amlogic TM2 T962E2 AB311";
@@ -83,10 +83,10 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
 			linux,contiguous-region;
-			alloc-ranges = <0x30000000 0x50000000>;
+			alloc-ranges = <0x0 0x30000000>;
 		};
 
 		/* codec shared reserved */
@@ -100,7 +100,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x8000000>;
+			size = <0xC800000>;
 			alignment = <0x400000>;
 			no-kernel-map;
 		};
@@ -151,7 +151,8 @@
 			 * 10x4736064=45.2M(0x2e) support 12bit
 			 * 10x4074560=40M(0x28) support 10bit
 			 */
-			size = <0x02800000>;
+			//size = <0x02800000>;
+			size = <0x08800000>;//2ch
 			alignment = <0x400000>;
 		};
 
@@ -211,9 +212,34 @@
 		status = "okay";
 	};
 
+	di_multi_v3 {
+		compatible = "amlogic, dimv3-g12a";
+		status = "okay";
+		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
+		flag_cma = <1>;
+		//memory-region = <&di_reserved>;
+		memory-region = <&di_cma_reserved>;
+		interrupts = <0 46 1
+				0 40 1>;
+		interrupt-names = "pre_irq", "post_irq";
+		clocks = <&clkc CLKID_VPU_CLKB_TMP_COMP>,
+			<&clkc CLKID_VPU_CLKB_COMP>;
+		clock-names = "vpu_clkb_tmp_composite",
+			"vpu_clkb_composite";
+		clock-range = <334 667>;
+		/* buffer-size = <3621952>;(yuv422 8bit) */
+		//no use buffer-size = <4074560>;/*yuv422 fullpack*/
+		/* reserve-iomap = "true"; */
+		/* if enable nr10bit, set nr10bit-support to 1 */
+		post-wr-support = <1>;
+		nr10bit-support = <1>;
+		nrds-enable = <1>;
+		pps-enable = <1>;
+		ch_nub = <2>;
+	};
 	deinterlace {
 		compatible = "amlogic, deinterlace";
-		status = "okay";
+		status = "disable";
 		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
 		flag_cma = <1>;
 		//memory-region = <&di_reserved>;
@@ -1799,7 +1825,7 @@
 	//pinctrl-names = "internal_eth_pins", "internal_gpio_pins";
 	//pinctrl-0 = <&internal_eth_pins>;
 	//pinctrl-1 = <&internal_gpio_pins>;
-	mc_val = <0x4be04>;
+	mc_val = <0x49e04>;
 
 	internal_phy=<1>;
 };
diff --git a/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts
index e886aeba34eb..50023990e57a 100644
--- a/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts
@@ -19,7 +19,7 @@
 
 #include "mesontm2.dtsi"
 #include "mesontm2_drm.dtsi"
-#include "partition_mbox_normal_P_32.dtsi"
+#include "partition_tv_ab_rdk.dtsi"
 #include "mesontm2_t962x3_ab301-panel.dtsi"
 
 / {
@@ -92,10 +92,10 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
 			linux,contiguous-region;
-			alloc-ranges = <0x30000000 0x50000000>;
+			alloc-ranges = <0x0 0x30000000>;
 		};
 
 		/* codec shared reserved */
@@ -109,7 +109,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x2000000>;
+			size = <0xC800000>;
 			alignment = <0x400000>;
 			no-kernel-map;
 		};
@@ -160,7 +160,8 @@
 			 * 10x4736064=45.2M(0x2e) support 12bit
 			 * 10x4074560=40M(0x28) support 10bit
 			 */
-			size = <0x02800000>;
+			//size = <0x02800000>;
+			size = <0x08800000>;//2ch
 			alignment = <0x400000>;
 		};
 
@@ -219,9 +220,34 @@
 		status = "okay";
 	};
 
+	di_multi_v3 {
+		compatible = "amlogic, dimv3-g12a";
+		status = "okay";
+		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
+		flag_cma = <1>;
+		//memory-region = <&di_reserved>;
+		memory-region = <&di_cma_reserved>;
+		interrupts = <0 46 1
+				0 40 1>;
+		interrupt-names = "pre_irq", "post_irq";
+		clocks = <&clkc CLKID_VPU_CLKB_TMP_COMP>,
+			<&clkc CLKID_VPU_CLKB_COMP>;
+		clock-names = "vpu_clkb_tmp_composite",
+			"vpu_clkb_composite";
+		clock-range = <334 667>;
+		/* buffer-size = <3621952>;(yuv422 8bit) */
+		//no use buffer-size = <4074560>;/*yuv422 fullpack*/
+		/* reserve-iomap = "true"; */
+		/* if enable nr10bit, set nr10bit-support to 1 */
+		post-wr-support = <1>;
+		nr10bit-support = <1>;
+		nrds-enable = <1>;
+		pps-enable = <1>;
+		ch_nub = <2>;
+	};
 	deinterlace {
 		compatible = "amlogic, deinterlace";
-		status = "okay";
+		status = "disable";
 		/* 0:use reserved; 1:use cma; 2:use cma as reserved */
 		flag_cma = <1>;
 		//memory-region = <&di_reserved>;
@@ -1849,7 +1875,7 @@
 	pinctrl-names = "internal_eth_pins", "internal_gpio_pins";
 	pinctrl-0 = <&internal_eth_pins>;
 	pinctrl-1 = <&internal_gpio_pins>;
-	mc_val = <0x4be04>;
+	mc_val = <0x49e04>;
 
 	internal_phy=<1>;
 };
-- 
2.29.0

